checkov/common/graph/graph_builder/variable_rendering/vertex_reference.py

```diff
@@ -0,0 +1,27 @@
+from typing import Union, List, Any
+from checkov.common.graph.graph_builder.graph_components.block_types import BlockType
+from abc import ABC, abstractmethod
+
+
+class TerraformVertexReference(ABC):
+    def __init__(self, block_type: Union[str, BlockType], sub_parts: List[str], origin_value: str) -> None:
+        self.block_type: BlockType = self.block_type_str_to_enum(block_type) if isinstance(block_type, str) else block_type
+        self.sub_parts = sub_parts
+        self.origin_value = origin_value
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, TerraformVertexReference):
+            return NotImplemented
```

Comment:
this is more a design decision, should the code really raise an error or just return `False`